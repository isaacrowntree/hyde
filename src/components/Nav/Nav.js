import React, { Component } from 'react';
import { LOGOUT, Store } from './../../Redux';

class Nav extends Component {
  constructor(props) {
    super(props);

    this.logout = this.logout.bind(this);
  }

  logout(e) {
    Store.dispatch({type: LOGOUT});
    e.preventDefault();
  }

  render() {
    return (

      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <a className="navbar-brand" href="#">Hyde Web Editor</a>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className="nav-item active">
              <a className="nav-link" href="#" onClick={this.logout}>Logout</a>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default Nav;
