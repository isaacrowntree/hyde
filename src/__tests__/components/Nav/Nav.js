import React from 'react';
import renderer from 'react-test-renderer';
import Nav from '../../../components/Nav/Nav';

test('Nav renders correctly', () => {
  const tree = renderer
    .create(<Nav />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
