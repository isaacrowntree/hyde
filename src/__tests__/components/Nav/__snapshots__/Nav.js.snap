// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<nav
  className="navbar navbar-light bg-light"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Hyde Web Editor
  </a>
  <div
    className=""
    id="navbarNav"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Logout
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
